### DISCUSSION TITLE
Trying to work out how to configure codeql to recognise different filename extensions for a language

### DISCUSSION THREAD

**Discussion by ThosRTanner** [community/community#57642] [Code Security] [answered]
### Select Topic Area

Question

### Body

What the title says, basically.

I have a repository where a lot of the javascript files have .jsm extensions, and it would be problematic to change them (and it helps me see that they're pre-es6 modules).

I can't find any way of persuading codeql that these are legit javascript files and it'd be really great if I could. It doesn't seem to be an unusual thing to support user specified extensions to identify supported languages in other products.

**Comment by jketema** [comment, marked as answer]
Hi @ThosRTanner,

Thanks for your question. There is a -not very well documented - environment variable `LGTM_INDEX_FILETYPES` that you can set to achieve this. In your case you should be to set `LGTM_INDEX_FILETYPES=".jsm:js"`.

For some very sparse details see: https://github.com/github/codeql/blob/467231e46940d3b9d3536c05509ce536e9bf9718/javascript/extractor/src/com/semmle/js/extractor/AutoBuild.java#L96-L98

**Comment by ThosRTanner** [comment]
Hmm. I've tried this: https://github.com/ThosRTanner/inforss/blob/master/.github/workflows/codeql.yml

It certainly has an effect as its now attempting to scan 68 files. However, it failed with an error

```
Encountered a fatal error while running "/opt/hostedtoolcache/CodeQL/2.14.1/x64/codeql/codeql database trace-command -O=javascript.trap.cache.dir=/home/runner/work/_temp/trapCaches/javascript -O=javascript.trap.cache.bound=1024 -O=javascript.trap.cache.write=true /home/runner/work/_temp/codeql_databases/javascript -- /opt/hostedtoolcache/CodeQL/2.14.1/x64/codeql/javascript/tools/autobuild.sh". Exit code was 2 and error was: A fatal error occurred: Exit status 1 from command: [/opt/hostedtoolcache/CodeQL/2.14.1/x64/codeql/javascript/tools/autobuild.sh]
```

**Reply by jketema** [reply]
Do you have a link to the workflow that shows this error?

**Reply by ThosRTanner** [reply]
i think it must have somehow got out of sync with what i had in the repo. Or it was showing me the wrong run. I force reran it by doing a small no-effect commit and it worked fine.

**Reply by jketema** [reply]
Good to hear. Thanks for letting me know.
